src/metabase/native_query_analyzer.clj

```diff
@@ -0,0 +1,82 @@
+(ns metabase.native-query-analyzer
+  "Integration with Macaw, which parses native SQL queries. All SQL-specific logic is in Macaw, the purpose of this
+  namespace is to:
+
+  1. Translate Metabase-isms into generic SQL that Macaw can understand.
+  2. Contain Metabase-specific business logic."
+  (:require
+   [clojure.string :as str]
+   [macaw.core :as mac]
+   [metabase.config :as config]
+   [metabase.util :as u]
+   [metabase.util.log :as log]
+   [toucan2.core :as t2])
+  (:import
+   [net.sf.jsqlparser JSQLParserException]))
+
+(def ^:dynamic *parse-queries-in-test?*
+  "Normally, a native card's query is parsed on every create/update. For most tests, this is an unnecessary
+  expense. Therefore, we skip parsing while testing unless this variable is turned on.
+
+  c.f. [[active?]]"
+  false)
+
+(defn- active?
+  "Should the query run? Either we're not testing or it's been explicitly turned on.
+
+  c.f. [[*parse-queries-in-test?*]]"
+  []
+  (or (not config/is-test?)
+      *parse-queries-in-test?*))
+
+(defn- normalize-name
+  ;; TODO: This is wildly naive and will be revisited once the rest of the plumbing is sorted out
+  ;; c.f. Milestone 3 of the epic: https://github.com/metabase/metabase/issues/36911
+  [name]
+  (-> name
+      (str/replace "\"" "")
+      u/lower-case-en))
+
+(defn- field-ids-for-query
+  "Very naively selects the IDs of Fields that could be used in the query. Improvements to this are planned for Q2 2024,
```

Comment:
I was a bit surprised by how naive this is; I think that's fine for now, but we should make it clearer so people who want to use this are aware of this fact.

How about documenting it in this NS's docstring?